### PR TITLE
Moadeli/temp sycl mp fixes

### PR DESCRIPTION
**Pull request by mmoadeli**
This PR  covers temporary changes to make Polygeist process SYCL input files. 
The fixes provided need serious re-visit to be addressed properly through collaboration with Polygeist.

### CODE REVIEW

File: mlir-clang/Lib/clang-mlir.cc [hunk 1]
```diff
@@ -593,6 +595,10 @@ ValueCategory MLIRScanner::VisitVarDecl(clang::VarDecl *decl) {
         subType = inite.val.getType();
       }
     }
+    else {
+      LLVMABI = false;
+      memtype = 5;
```
Comment (rioffe) [inline mlir-clang/Lib/clang-mlir.cc:600]: Sorry, what is the magic number 5 here? Could you please use enum or constant with a meaningful name?

File: mlir-clang/Lib/clang-mlir.cc [hunk 2]
```diff
@@ -483,9 +483,9 @@ static void initializeValueByInitListExpr(mlir::Value toInit, clang::Expr *expr,
   // memref.store operations. This requires that the memref value should
   // have static shape.
   MemRefType memTy = toInit.getType().cast<MemRefType>();
-  assert(memTy.hasStaticShape() &&
-         "The memref to be initialized by InitListExpr should have static "
-         "shape.");
+  // assert(memTy.hasStaticShape() &&
```
Comment (rioffe) [inline mlir-clang/Lib/clang-mlir.cc:486]: Let's get rid of the dead code here

File: mlir-clang/Lib/clang-mlir.cc [hunk 3]
```diff
@@ -689,15 +695,15 @@ ValueCategory MLIRScanner::VisitInitListExpr(clang::InitListExpr *expr) {
     return ValueCategory(op, true);
   }
   if (expr->getNumInits()) {
-    assert(op.getType().cast<MemRefType>().getShape().size() <= 2);
-    assert((op.getType().cast<MemRefType>().getShape().size() == 1 &&
-            op.getType().cast<MemRefType>().getShape()[0] == -1) ||
-           op.getType().cast<MemRefType>().getShape().back() ==
-               expr->getNumInits());
+    // assert(op.getType().cast<MemRefType>().getShape().size() <= 2);
```
Comment (rioffe) [inline mlir-clang/Lib/clang-mlir.cc:698]: Let's get rid of the dead code here

File: mlir-clang/Lib/clang-mlir.cc [hunk 4]
```diff
@@ -689,15 +695,15 @@ ValueCategory MLIRScanner::VisitInitListExpr(clang::InitListExpr *expr) {
...
+    // assert((op.getType().cast<MemRefType>().getShape().size() == 1 &&
+    //         op.getType().cast<MemRefType>().getShape()[0] == -1) ||
+    //        op.getType().cast<MemRefType>().getShape().back() ==
+    //            expr->getNumInits());
     for (size_t i = 0; i < expr->getNumInits(); ++i) {
       auto inite = Visit(expr->getInit(i)).getValue(builder);
-      assert(inite.getType() ==
-             op.getType().cast<MemRefType>().getElementType());
+      // assert(inite.getType() ==
```
Comment (rioffe) [inline mlir-clang/Lib/clang-mlir.cc:705]: Let's get rid of the dead code here

File: mlir-clang/Lib/clang-mlir.cc [hunk 5]
```diff
@@ -5399,6 +5410,13 @@ void MLIRASTConsumer::run() {
 
     if (done.count(name))
       continue;
+
+    if (StringRef(name).startswith("_Z28__"))
```
Comment (rioffe) [inline mlir-clang/Lib/clang-mlir.cc:5414]: Is this specific type of mangling you are looking for? What type is it correspond to?

File: mlir-clang/Lib/clang-mlir.cc [hunk 6]
```diff
@@ -6153,8 +6179,11 @@ static bool parseMLIR(const char *Argv0, std::vector<std::string> filenames,
...
...6_64-linux-gnu/c++/9", "-internal-isystem", "/usr/lib/gcc/x86_64-linux-gnu/9/../../../../include/c++/9/backward", "-internal-isystem", "/home/mahmoud/codeplay/Polygeist/llvm-project/build/lib/clang/14.0.0/include", "-internal-isystem", "/usr/local/include", "-internal-isystem", "/usr/lib/gcc/x86_64-linux-gnu/9/../../../../x86_64-linux-gnu/include", "-internal-externc-isystem", "/usr/include/x86_64-linux-gnu", "-internal-externc-isystem", "/include", "-internal-externc-isystem", "/usr/include", "-internal-isystem", "/home/mahmoud/codeplay/Polygeist/llvm-project/build/lib/clang/14.0.0/include", "-internal-isystem", "/usr/local/include", "-internal-isystem", "/usr/lib/gcc/x86_64-linux-gnu/9/../../../../x86_64-linux-gnu/include", "-internal-externc-isystem", "/usr/include/x86_64-linux-gnu", "-internal-externc-isystem", "/include", "-internal-externc-isystem", "/usr/include", "-std=c++17", "-fdeprecated-macro", "-fdebug-compilation-dir=/home/mahmoud/codeplay/Polygeist/llvm-project/build/test", "-ferror-limit", "19", "-fgnuc-version=4.2.1", "-fcxx-exceptions", "-fexceptions", "-fcolor-diagnostics", "-faddrsig", "-D__GCC_HAVE_DWARF2_CFI_ASM=1", "-o", "/tmp/no-loop-c59c82.bc", "-x", "c++",
```
Comment (rioffe) [inline mlir-clang/Lib/clang-mlir.cc:6182]: It is unfortunate that this is hardcoded to your home directory. Anything that can be done here about it?

File: mlir-clang/Lib/clang-mlir.cc [hunk 7]
```diff
@@ -6213,11 +6242,14 @@ static bool parseMLIR(const char *Argv0, std::vector<std::string> filenames,
         StringAttr::get(module->getContext(),
                         Clang->getTarget().getTriple().getTriple()));
 
-    for (const auto &FIF : Clang->getFrontendOpts().Inputs) {
+//    for (const auto &FIF : Clang->getFrontendOpts().Inputs)
```
Comment (rioffe) [inline mlir-clang/Lib/clang-mlir.cc:6245]: delete?

File: mlir-clang/mlir-clang.cc [hunk 8]
```diff
@@ -441,14 +441,14 @@ int main(int argc, char **argv) {
...
     }
-    if (mlir::failed(pm.run(module.get()))) {
-      module->dump();
-      return 4;
-    }
-    if (mlir::failed(mlir::verify(module.get()))) {
-      module->dump();
-      return 5;
-    }
+    // if (mlir::failed(pm.run(module.get()))) {
```
Comment (rioffe) [inline mlir-clang/mlir-clang.cc:444]: delete?

File: mlir-clang/mlir-clang.cc [hunk 9]
```diff
@@ -411,7 +411,7 @@ int main(int argc, char **argv) {
   bool LinkOMP = false;
   pm.enableVerifier(EarlyVerifier);
   mlir::OpPassManager &optPM = pm.nest<mlir::FuncOp>();
-  if (true) {
+  if (false) {
```
Comment (rioffe) [inline mlir-clang/mlir-clang.cc:414]: Do you still need this code?

**Review by rioffe — COMMENTED** [review]
See my comments: mostly cosmetic

File: mlir-clang/Lib/clang-mlir.cc (hunk 1 above)
Comment (mmoadeli) [reply mlir-clang/Lib/clang-mlir.cc:600]: This work is provided just to be able to generate MLIR ftom SYCL. There are lot's of work around. A better fix is in https://github.com/wsmoses/Polygeist/pull/115

File: mlir-clang/Lib/clang-mlir.cc (hunk 2 above)
Comment (mmoadeli) [reply mlir-clang/Lib/clang-mlir.cc:486]: This is reverted in https://github.com/wsmoses/Polygeist/pull/115.

File: mlir-clang/Lib/clang-mlir.cc (hunk 3 above)
Comment (mmoadeli) [reply mlir-clang/Lib/clang-mlir.cc:698]: This is reverted in https://github.com/wsmoses/Polygeist/pull/115

File: mlir-clang/Lib/clang-mlir.cc (hunk 4 above)
Comment (mmoadeli) [reply mlir-clang/Lib/clang-mlir.cc:705]: This is improved in https://github.com/wsmoses/Polygeist/pull/115

File: mlir-clang/Lib/clang-mlir.cc (hunk 5 above)
Comment (mmoadeli) [reply mlir-clang/Lib/clang-mlir.cc:5414]: This corresponded to some spir-v function. This was too low level to be important for our use case.

File: mlir-clang/Lib/clang-mlir.cc (hunk 6 above)
Comment (mmoadeli) [reply mlir-clang/Lib/clang-mlir.cc:6182]: this is addressed in https://github.com/InteonCo/Polygeist/pull/2

File: mlir-clang/Lib/clang-mlir.cc (hunk 7 above)
Comment (mmoadeli) [reply mlir-clang/Lib/clang-mlir.cc:6245]: addressed in https://github.com/InteonCo/Polygeist/pull/2.

File: mlir-clang/mlir-clang.cc (hunk 8 above)
Comment (mmoadeli) [reply mlir-clang/mlir-clang.cc:444]: this needs to be reverted.

File: mlir-clang/mlir-clang.cc (hunk 9 above)
Comment (mmoadeli) [reply mlir-clang/mlir-clang.cc:414]: this needs to be reverted.

**Comment by mmoadeli** [comment]
No need for this anymore

**Pull request closed without merging**